### PR TITLE
Delete usage status #251

### LINKED ISSUE #251: Implement API Endpoint for Deleting Usage Statuses
- Check if the usage status exist in the items collection

### PR DESCRIPTION
**Pull request by joshuadkitenge**
## Description
- Checks if the usage_status_id is referenced in the items collection and if it is then allows delete. if not it errors  

## Testing instructions
- [ ] Review code
- [ ] Check Actions build

## Agile board tracking
closes #251

### CODE REVIEW

File: inventory_management_system_api/repositories/usage_status.py
Comment (joelvdavies) [inline inventory_management_system_api/repositories/usage_status.py:90]:
```suggestion
        Delete a usage status by its ID from MongoDB database.
        
        Checks if usage status is a part of an item, and does not delete if it is
```

File: inventory_management_system_api/services/usage_status.py
Comment (joelvdavies) [inline inventory_management_system_api/services/usage_status.py:55]:
```suggestion
        Delete a usage status by its ID
        
        :param usage_status_id: The ID of the usage status to delete
```

File: inventory_management_system_api/repositories/usage_status.py
Comment (joelvdavies) [inline inventory_management_system_api/repositories/usage_status.py:116]:
```suggestion
    def _is_usage_status_in_item(self, usage_status_id: str, session: ClientSession = None) -> bool:
```

File: inventory_management_system_api/routers/v1/usage_status.py
Comment (joelvdavies) [inline inventory_management_system_api/routers/v1/usage_status.py:98]:
```suggestion
        message = "Usage status not found"
        logger.exception(message)
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=message) from exc
```
For consistency with others (it seems manufacturer is the exception).

File: inventory_management_system_api/routers/v1/usage_status.py
Comment (joelvdavies) [inline inventory_management_system_api/routers/v1/usage_status.py:103]:
```suggestion
        message = "The specified usage status is a part of an Item"
        logger.exception(message)
        raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail=message) from exc
```

File: test/e2e/test_usage_status.py
Comment (joelvdavies) [inline test/e2e/test_usage_status.py:186]:
```suggestion
    assert response.json()["detail"] == "The specified usage status is a part of an Item"
```

File: test/unit/repositories/test_usage_status.py
Comment (joelvdavies) [inline test/unit/repositories/test_usage_status.py:229]:
```suggestion
def test_delete_with_a_non_existent_id(
```

File: test/unit/repositories/test_usage_status.py
Comment (joelvdavies) [inline test/unit/repositories/test_usage_status.py:273]:
```suggestion
        == f"The usage status with id {str(usage_status_id)} is a part of an Item"
```

**Review by joelvdavies — APPROVED** [review]
LGTM, don't merge until dev is migrated and functioning properly

File: inventory_management_system_api/core/exceptions.py
Comment (VKTB) [inline inventory_management_system_api/core/exceptions.py:75]:
```suggestion
    Exception raised when attempting to delete a usage status that is a part of an item
```

File: inventory_management_system_api/routers/v1/usage_status.py
Comment (VKTB) [inline inventory_management_system_api/routers/v1/usage_status.py:89]:
```suggestion
def delete_usage_status(usage_status_id: str, usage_status_service: Annotated[UsageStatusService, Depends(UsageStatusService)]) -> None:
```

File: inventory_management_system_api/repositories/usage_status.py
Comment (VKTB) [inline inventory_management_system_api/repositories/usage_status.py:84]: Docstring should be updated to include the `session` param.

File: inventory_management_system_api/repositories/usage_status.py
Comment (VKTB) [inline inventory_management_system_api/repositories/usage_status.py:96]:
```suggestion
            raise PartOfItemError(f"The usage status with ID {str(usage_status_id)} is a part of an Item")
```

File: test/e2e/test_usage_status.py
Comment (VKTB) [inline test/e2e/test_usage_status.py:135]:
```suggestion
    """Test trying to delete a usage status that is a part of an Item"""
```

File: inventory_management_system_api/repositories/usage_status.py
Comment (VKTB) [inline inventory_management_system_api/repositories/usage_status.py:115]: I think this should say
```suggestion
        """Checks to see if any of the items in the database have a specific usage status ID
```

File: inventory_management_system_api/repositories/usage_status.py
Comment (VKTB) [inline inventory_management_system_api/repositories/usage_status.py:119]:
```suggestion
        :return: `True` if 1 or more items have the usage status ID, `False` otherwise
```

**Review by VKTB — CHANGES_REQUESTED** [review]

**Review by joelvdavies — DISMISSED** [review]

**Review by VKTB — DISMISSED** [review]
Probably squash and merge this one too when it is time to be merged

File: inventory_management_system_api/routers/v1/usage_status.py
Comment (VKTB) [inline inventory_management_system_api/routers/v1/usage_status.py:91]:
```suggestion
def delete_usage_status(
    usage_status_id: Annotated[str, Path(description="ID of the usage status to delete")],
    usage_status_service: UsageStatusService = Depends()
) -> None:
```

Comment (joelvdavies) [reply inventory_management_system_api/routers/v1/usage_status.py:91]: I would still use `usage_status_service: Annotated[UsageStatusService, Depends(UsageStatusService)]` as its preferred in the docs #117

Comment (VKTB) [reply inventory_management_system_api/routers/v1/usage_status.py:91]: Sorry, that was a copy and paste error. Good spot, thanks!

**Review by joelvdavies — APPROVED** [review]

**Review by VKTB — APPROVED** [review]

**Pull request merged**
